### PR TITLE
Datseris runnable docs

### PR DESCRIPTION
**Pull request by SimonDanisch**
@datseris, I can't push to your branch, so I created a new PR... If you want to continue working on this PR, I can give you access..

### CODE REVIEW

**Comment by Datseris** [comment]
Now you have made it so both `yautolimits = false` but also we do not set limits after axis creation. So how could the axis possibly get correct limits?

**Comment by Datseris** [comment]
> If you want to continue working on this PR, I can give you access..

Yes please!

**Comment by Datseris** [comment]
For reference, what will be left for future PRs (will translate this into issues once we merge the PR)
- [ ] Better automatic axis ticks (for better support for local plots) 
- [ ] Function for "antimeridian cutting"
- [ ] Coastlines plot uses antimeridian cutting
- [ ] Add inverse transform from Pro4 (@visr )
- [ ] Add actual, proper tests

**Comment by SimonDanisch** [comment]
>none of them now seem to be plotted correctly.

Yeah we need to wait for the new Makie release, then the usual axis autolimits will do the trick.

>Now you have made it so both yautolimits = false

Ah that was just an experiment...It gets ignored anyways, since its actually a plot attribute...

**Comment by Datseris** [comment]
> Yeah we need to wait for the new Makie release, then the usual axis autolimits will do the trick.

Well then I really hope this comes soon so we can finish this!

File: src/geoaxis.jl [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+## Example
+```julia
+using GeoMakie, GLMakie
+
+lons = -180:180
+lats = -90:90
+field = [exp(cosd(l)) + 3(y/90) for l in lons, y in lats]
+
+# Plot coastlines
+coastplot = lines!(ax, GeoMakie.coastlines(); color = :black, overdraw = true, coastkwargs...)
```
Comment (simonbyrne) [inline src/geoaxis.jl:34]: `coastkwargs` is not defined
```suggestion
coastplot = lines!(ax, GeoMakie.coastlines(); color = :black, overdraw = true)
```

Comment (simonbyrne) [reply src/geoaxis.jl:34]: Also, `ax` is not defined

File: src/geoaxis.jl [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+translate!(coastplot, 0, 0, 99) # ensure they are on top of other plotted elements
+
+# Surface example
+fig = Figure()
+ax = GeoAxis(fig[1,1])
+surface!(ax, lons, lats, field)
+display(fig)
```
Comment (simonbyrne) [inline src/geoaxis.jl:41]: I get the following error:
```
Error in callback:
MethodError: no method matching inverse_transform(::Makie.PointTrans{2, Proj4.Transformation})
Closest candidates are:
  inverse_transform(::Tuple) at ~/.julia/packages/Makie/gQOQF/src/layouting/transformation.jl:360
  inverse_transform(::typeof(identity)) at ~/.julia/packages/Makie/gQOQF/src/layouting/transformation.jl:354
  inverse_transform(::typeof(log10)) at ~/.julia/packages/Makie/gQOQF/src/layouting/transformation.jl:355
  ...
Stacktrace:
  [1] process_interaction(r::Makie.MakieLayout.RectangleZoom, event::MouseEvent, ax::Axis)
    @ Makie.MakieLayout ~/.julia/packages/Makie/gQOQF/src/makielayout/interactions.jl:150
  [2] (::Makie.MakieLayout.var"#process_event#312"{Axis})(event::MouseEvent)
    @ Makie.MakieLayout ~/.julia/packages/Makie/gQOQF/src/makielayout/layoutables/axis.jl:410
  [3] #invokelatest#2
    @ ./essentials.jl:716 [inlined]
  [4] invokelatest
    @ ./essentials.jl:714 [inlined]
  [5] (::Makie.PrioCallback)(val::MouseEvent)
    @ Makie ~/.julia/packages/Makie/gQOQF/src/interaction/PriorityObservable.jl:17
  [6] #invokelatest#2
    @ ./essentials.jl:716 [inlined]
  [7] invokelatest
    @ ./essentials.jl:714 [inlined]
  [8] notify(observable::Makie.PriorityObservable{MouseEvent})
    @ Makie ~/.julia/packages/Makie/gQOQF/src/interaction/PriorityObservable.jl:85
  [9] setindex!
    @ ~/.julia/packages/Makie/gQOQF/src/interaction/PriorityObservable.jl:79 [inlined]
 [10] (::Makie.MakieLayout.var"#99#101"{Scene, Makie.MakieLayout.var"#is_mouse_over_relevant_area#96"{Scene, Tuple{}}, Base.RefValue{Float64}, Base.RefValue{Bool}, Base.RefValue{Bool}, Base.RefValue{Union{Nothing, Makie.Mouse.Button}}, Base.RefValue{Bool}, Base.RefValue{GeometryBasics.Point{2, Float32}}, Base.RefValue{GeometryBasics.Point{2, Float32}}, Base.RefValue{Makie.Mouse.Action}, Makie.PriorityObservable{MouseEvent}, Module})(mp::Tuple{Float64, Float64})
    @ Makie.MakieLayout ~/.julia/packages/Makie/gQOQF/src/makielayout/mousestatemachine.jl:211
 [11] #invokelatest#2
    @ ./essentials.jl:716 [inlined]
 [12] invokelatest
    @ ./essentials.jl:714 [inlined]
 [13] (::Makie.PrioCallback)(val::Tuple{Float64, Float64})
    @ Makie ~/.julia/packages/Makie/gQOQF/src/interaction/PriorityObservable.jl:17
 [14] #invokelatest#2
    @ ./essentials.jl:716 [inlined]
 [15] invokelatest
    @ ./essentials.jl:714 [inlined]
 [16] notify(observable::Makie.PriorityObservable{Tuple{Float64, Float64}})
    @ Makie ~/.julia/packages/Makie/gQOQF/src/interaction/PriorityObservable.jl:85
 [17] setindex!
    @ ~/.julia/packages/Makie/gQOQF/src/interaction/PriorityObservable.jl:79 [inlined]
 [18] (::GLMakie.var"#69#70"{Events, GLFW.Window})(#unused#::Nothing)
    @ GLMakie ~/.julia/packages/GLMakie/pFGSp/src/events.jl:6
 [19] #invokelatest#2
    @ ./essentials.jl:716 [inlined]
 [20] invokelatest
    @ ./essentials.jl:714 [inlined]
 [21] notify
    @ ~/.julia/packages/Observables/OFj0u/src/Observables.jl:88 [inlined]
 [22] setindex!
    @ ~/.julia/packages/Observables/OFj0u/src/Observables.jl:248 [inlined]
 [23] fps_renderloop(screen::GLMakie.Screen, framerate::Float64)
    @ GLMakie ~/.julia/packages/GLMakie/pFGSp/src/rendering.jl:22
 [24] renderloop(screen::GLMakie.Screen; framerate::Float64)
    @ GLMakie ~/.julia/packages/GLMakie/pFGSp/src/rendering.jl:48
 [25] renderloop(screen::GLMakie.Screen)
    @ GLMakie ~/.julia/packages/GLMakie/pFGSp/src/rendering.jl:41
 [26] (::GLMakie.var"#50#52"{GLMakie.Screen})()
    @ GLMakie ./task.jl:423
```

Comment (simonbyrne) [reply src/geoaxis.jl:41]: Ah, that appears to be https://github.com/JuliaPlots/GeoMakie.jl/issues/77

Comment (Datseris) [reply src/geoaxis.jl:41]: Yeah this is known but Simon said its solved on master.

Comment (simonbyrne) [reply src/geoaxis.jl:41]: master of Makie?

**Comment by Datseris** [comment]
In my original version the GeoAxis by default plotted the coastlines. @lazarusA argued this shouldn't be done, so I accepted this change and now in current code GeoAxis does not plot coastlines by default.

Now, I think this was the wrong move. Simply put, the situations where you would want coastlines are much more frequent than those than you wouldn't want. So it makes sense to plot this by default. I would argue we should bring this change back. We don't have to allow keyword arguments for the coastline plot, just some basic defaults: black thin lines. If a user wants more control over coastlines then the user can simply tell GeoAxis to not plot coastlines.

**Comment by visr** [comment]
> Simply put, the situations where you would want coastlines are much more frequent than those than you wouldn't want.

For your domain that is probably true, but not for some other domains. I wouldn't want it most of the time. Though I'd appreciate easily being able to add it for sure. Others might argue for having country borders by default.

I'd really like this package to remain domain agnostic. E.g. planetary scientists should be able to plot maps of other bodies than Earth (PROJ supports that), as well as local plots.

**Comment by Datseris** [comment]
ok, sounds convincing to me.

**Comment by Datseris** [comment]
> Though I'd appreciate easily being able to add it for sure.

Well, we can easily satisfy both worlds: keyword argument `coastlines = false` that if true will plot the coastlines.

**Comment by SimonDanisch** [comment]
Ok, I think we can merge & tag this..
https://github.com/JuliaPlots/Makie.jl/pull/1574 will improve the camera & zooming behaviour, but isn't strictly needed.

**Comment by Datseris** [comment]
thanks simon. Okay, I'll do any documentation improvements I find in a separate PR. Thank you very much for your support, I really appreciate it :)

**Pull request merged**